packages/aws-cdk/lib/api/garbage-collection/garbage-collector.ts

```diff
@@ -311,11 +440,44 @@ export class GarbageCollector {
     debug(`Untagged ${untaggables.length} assets`);
   }
 
+  /**
+   * Tag images in parallel using p-limit
+   */
+  private async parallelTagEcr(ecr: ECR, repo: string, taggables: ImageAsset[], printer: ProgressPrinter) {
+    const limit = pLimit(P_LIMIT);
+
+    for (let i = 0; i < taggables.length; i++) {
+      const img = taggables[i];
+      const tagEcr = async () => {
+        const date = Date.now();
+        let imageTag;
+        try {
+          imageTag = `${ECR_ISOLATED_TAG}-${i}-${String(date)}`;
```

Comment:
Please turn these fields around, so that they more or less lexicographically sort correctly.

Also -- you have logic around building and splitting this string. Would be nice if that logic lived together. So the building logic should live next to the splitting logic, most likely on `ImageAsset` then.